src/cpp/src/whisper_pipeline_static.cpp

```diff
@@ -654,7 +626,11 @@ WhisperDecodedResults WhisperPipeline::StaticWhisperPipeline::generate(
 
         // prepare init_ids just once for whole input
         if (init_ids.empty()) {
+            m_models.decoder = m_decoder_cache.get_model(1).create_infer_request(); // for detect_language()
```

Comment:
Here the model compiled upfront again. What if it's not even needed and language already known?

I believe you need it somewhere here: https://github.com/openvinotoolkit/openvino.genai/blob/614e6d9b7802d89b00c480e20b27c0f5b7913b04/src/cpp/src/whisper_pipeline_static.cpp#L269